protocols/imap/src/test/java/org/apache/james/imap/processor/base/SelectedMailboxImplTest.java

```diff
@@ -0,0 +1,172 @@
+import java.util.concurrent.Executors;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import javax.mail.Flags;
+
+import org.apache.james.imap.api.ImapSessionUtils;
+import org.apache.james.imap.api.process.ImapSession;
+import org.apache.james.mailbox.MailboxListener;
+import org.apache.james.mailbox.MailboxManager;
+import org.apache.james.mailbox.MailboxSession;
+import org.apache.james.mailbox.MessageManager;
+import org.apache.james.mailbox.MessageUid;
+import org.apache.james.mailbox.model.MailboxConstants;
+import org.apache.james.mailbox.model.MailboxPath;
+import org.apache.james.mailbox.model.MessageMetaData;
+import org.apache.james.mailbox.model.SearchQuery;
+import org.apache.james.mailbox.store.SimpleMessageMetaData;
+import org.apache.james.mailbox.store.event.EventFactory;
+import org.apache.james.mailbox.store.mail.model.DefaultMessageId;
+import org.apache.james.mailbox.store.mail.model.Mailbox;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Ignore;
+import org.junit.Test;
+import org.mockito.invocation.InvocationOnMock;
+import org.mockito.stubbing.Answer;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.collect.ImmutableList;
+
+
+public class SelectedMailboxImplTest {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(SelectedMailboxImplTest.class);
+    public static final MessageUid MESSAGE_UID_5 = MessageUid.of(5);
+
+    private ExecutorService executorService;
+    private MailboxManager mailboxManager;
+    private MessageManager messageManager;
+    private MailboxPath mailboxPath;
+    private ImapSession imapSession;
+    private Mailbox mailbox;
+
+    @Before
+    public void setUp() throws Exception {
+        executorService = Executors.newFixedThreadPool(1);
+
+        mailboxPath = new MailboxPath(MailboxConstants.USER_NAMESPACE, "tellier@linagora.com", MailboxConstants.INBOX);
+        mailboxManager = mock(MailboxManager.class);
+        messageManager = mock(MessageManager.class);
+        when(mailboxManager.getMailbox(eq(mailboxPath), any(MailboxSession.class))).thenReturn(messageManager);
+        when(messageManager.getApplicableFlags(any(MailboxSession.class))).thenReturn(new Flags());
+        when(messageManager.search(any(SearchQuery.class), any(MailboxSession.class)))
+            .then(sleepThenSearchAnswer());
+
+        imapSession = mock(ImapSession.class);
+        when(imapSession.getAttribute(ImapSessionUtils.MAILBOX_SESSION_ATTRIBUTE_SESSION_KEY)).thenReturn(mock(MailboxSession.class));
+
+        mailbox = mock(Mailbox.class);
+        when(mailbox.generateAssociatedPath()).thenReturn(mailboxPath);
+    }
+
+    @After
+    public void tearDown() {
+        executorService.shutdownNow();
+    }
+
+    @Ignore
+    @Test
+    public void concurrentEventShouldNotSkipUidEmmitedDuringInitialization() throws Exception {
+        final AtomicInteger success = new AtomicInteger(0);
```

Comment:
Ok you declare a variable with the name success letting the reader think you are going to check it's value but you don't. IMO this is a bit confusing. Is it one purpose